### PR TITLE
[AArch64] Update target feature requirements of SVE bfloat instructions

### PR DESCRIPTION
**Pull request by momchil-velikov**
According to the latest update of the ISA
https://developer.arm.com/documentation/ddi0602/2023-09/?lang=en all of the affected instruction encodings now require

    (FEAT_SVE2 or FEAT_SME2) and FEAT_SVE_B16B16

### CODE REVIEW

File: clang/include/clang/Basic/arm_sve.td [hunk 1]
```diff
@@ -2066,7 +2066,7 @@ let TargetGuard = "sve2p1|sme2" in {
   def SVPFALSE_COUNT_ALIAS : SInst<"svpfalse_c", "}v", "", MergeNone, "", [IsOverloadNone, IsStreamingCompatible]>;
 }
 
-let TargetGuard = "sve2p1,b16b16" in {
+let TargetGuard = "(sve2|sme2),b16b16" in {
```
Comment (sdesmalen-arm) [inline clang/include/clang/Basic/arm_sve.td:2069]: nit: Github doesn't allow me to select the right line, but the comment on line 2083 is no longer correct.

**Comment by momchil-velikov** [comment]
Rebased the clear the test run.

File: clang/include/clang/Basic/arm_sve.td [hunk 2]
```diff
@@ -2086,7 +2086,7 @@ let TargetGuard = "sve2p1|sme2" in {
   def SVCNTP_COUNT : SInst<"svcntp_{d}", "n}i", "QcQsQiQl", MergeNone, "aarch64_sve_cntp_{d}", [IsOverloadNone, IsStreamingCompatible], [ImmCheck<1, ImmCheck2_4_Mul2>]>;
 }
 
-let TargetGuard = "sve2p1,b16b16" in {
+let TargetGuard = "(sve2|sme2),b16b16" in {
```
Comment (CarolineConcatto) [inline clang/include/clang/Basic/arm_sve.td:2089]: If we are adding for sme2 then we need to add the streaming compatible flag and also update the tests.

Comment (momchil-velikov) [reply clang/include/clang/Basic/arm_sve.td:2089]: Yeah, I'll add the `IsStreamingCompatible` flag.

What needs to be updated in tests?

Comment (CarolineConcatto) [reply clang/include/clang/Basic/arm_sve.td:2089]: I do believe we should add:
__arm_streaming_compatible; 
in the tests.

svbfloat16_t test_svadd_bf16_m(svbool_t pg, svbfloat16_t op1, svbfloat16_t op2)   __arm_streaming_compatible
{
  return SVE_ACLE_FUNC(svadd, _bf16, _m)(pg, op1, op2);
}

Comment (momchil-velikov) [reply clang/include/clang/Basic/arm_sve.td:2089]: Done.

**Review by CarolineConcatto — APPROVED** [review]

**Pull request merged**